blackjax/inference/metrics.py

```diff
@@ -101,7 +131,37 @@ def kinetic_energy(momentum: PyTree, *_) -> float:
             velocity = jnp.matmul(inverse_mass_matrix, momentum)
             return 0.5 * jnp.dot(velocity, momentum)
 
-        return momentum_generator, kinetic_energy
+        def is_turning(
```

Comment:
Seems we can simplify the code path a bit here to do reduce duplicate code with:
```python
    if ndim == 1:  # diagonal mass matrix
        mass_matrix_sqrt = jnp.sqrt(jnp.reciprocal(inverse_mass_matrix))
        multiply_op = jnp.multiply
    elif ndim == 2:
        tril_inv = jscipy.linalg.cholesky(inverse_mass_matrix)
        identity = jnp.identity(shape[0])
        mass_matrix_sqrt = jscipy.linalg.solve_triangular(
            tril_inv, identity, lower=True
        )
        multiply_op = jnp.matmul
    def momentum_generator(rng_key: jax.random.PRNGKey, position: PyTree) -> PyTree:
        _, unravel_fn = ravel_pytree(position)
        std = jax.random.normal(rng_key, shape)
        momentum = multiply_op(std, mass_matrix_sqrt)
        return unravel_fn(momentum)
    def kinetic_energy(momentum: PyTree, *_) -> float:
        momentum, _ = ravel_pytree(momentum)
        momentum = jnp.array(momentum)
        velocity = multiply_op(inverse_mass_matrix, momentum)
        return 0.5 * jnp.dot(velocity, momentum)
```

Reply:
Yes, @canyon289 implemented a similar simplification in another PR. I'll rebase before merging.